src/e2e/java/teammates/e2e/util/BackDoor.java

```diff
@@ -425,6 +440,151 @@ public static StudentAttributes getStudent(String courseId, String studentEmail)
+        FeedbackSessionData sessionData = getFeedbackSessionData(courseId, feedbackSessionName);
+
+        FeedbackSessionAttributes sessionAttributes = FeedbackSessionAttributes
+                .builder(sessionData.getFeedbackSessionName(), sessionData.getCourseId())
+                .withInstructions(sessionData.getInstructions())
+                .withStartTime(Instant.ofEpochMilli(sessionData.getSubmissionStartTimestamp()))
+                .withEndTime(Instant.ofEpochMilli(sessionData.getSubmissionEndTimestamp()))
+                .withTimeZone(ZoneId.of(sessionData.getTimeZone()))
+                .withGracePeriod(Duration.ofMinutes(sessionData.getGracePeriod()))
+                .withIsClosingEmailEnabled(sessionData.getIsClosingEmailEnabled())
+                .withIsPublishedEmailEnabled(sessionData.getIsPublishedEmailEnabled())
+                .build();
+
+        sessionAttributes.setCreatedTime(Instant.ofEpochMilli(sessionData.getCreatedAtTimestamp()));
+
+        if (sessionData.getSessionVisibleSetting().equals(SessionVisibleSetting.AT_OPEN)) {
+            sessionAttributes.setSessionVisibleFromTime(Const.TIME_REPRESENTS_FOLLOW_OPENING);
+        } else {
+            sessionAttributes.setSessionVisibleFromTime(Instant.ofEpochMilli(
+                    sessionData.getCustomSessionVisibleTimestamp()));
+        }
+
+        if (sessionData.getResponseVisibleSetting().equals(ResponseVisibleSetting.AT_VISIBLE)) {
+            sessionAttributes.setResultsVisibleFromTime(Const.TIME_REPRESENTS_FOLLOW_VISIBLE);
+        } else if (sessionData.getResponseVisibleSetting().equals(ResponseVisibleSetting.LATER)) {
+            sessionAttributes.setResultsVisibleFromTime(Const.TIME_REPRESENTS_LATER);
+        } else {
+            sessionAttributes.setResultsVisibleFromTime(Instant.ofEpochMilli(
+                    sessionData.getCustomResponseVisibleTimestamp()));
+        }
+
+        return sessionAttributes;
+    }
+
+    /**
+     * Get soft deleted feedback session from datastore.
+     */
+    public static FeedbackSessionAttributes getSoftDeletedSession(String feedbackSessionName, String instructorId) {
+        Map<String, String[]> params = new HashMap<>();
+        params.put(Const.ParamsNames.ENTITY_TYPE, new String[] { Const.EntityType.INSTRUCTOR });
+        params.put(Const.ParamsNames.IS_IN_RECYCLE_BIN, new String[] { "true" });
+        params.put(Const.ParamsNames.USER_ID, new String[] { instructorId });
+        ResponseBodyAndCode response = executeGetRequest(Const.ResourceURIs.SESSIONS, params);
+        if (response.responseCode == HttpStatus.SC_NOT_FOUND) {
+            return null;
+        }
+
+        FeedbackSessionsData sessionsData = JsonUtils.fromJson(response.responseBody, FeedbackSessionsData.class);
+        List<FeedbackSessionData> sessionsDataList = sessionsData.getFeedbackSessions();
+        for (FeedbackSessionData feedbackSession : sessionsDataList) {
```

Comment:
Minor point:

I realise this is a common pattern within the `BackDoor`. I am personally not a fan of sweeping through a list to find and return an item inside. The happy path is not very clear either :P 

This is an alternate way of doing it:
```
FeedbackSessionData feedbackSession =  
    sessionsData.getFeedbackSessions()
                .stream()
                .filter(fs -> fs.getFeedbackSessionName().equals(feedbackSessionName))
                .findFirst()
                .orElse(null);

if (feedbackSession == null) {
    return null;
}

return FeedbackSessionAttributes
        .builder(feedbackSession.getCourseId(), 
                 feedbackSession.getFeedbackSessionName())
        .build();
```